src/pps.js

```diff
@@ -0,0 +1,58 @@
+import { auctionManager } from './auctionManager.js';
+import { TARGETING_KEYS } from './constants.js';
+import { CLIENT_SECTIONS } from './fpd/oneClient.js';
+import { deepAccess, uniques } from './utils.js';
+
+export function getSegments(fpd, sections, segtax) {
+  return sections
+    .flatMap(section => deepAccess(fpd, section) || [])
+    .filter(datum => datum.ext?.segtax === segtax)
+    .flatMap(datum => datum.segment?.map(seg => seg.id))
+    .filter(ob => ob)
+    .filter(uniques)
+}
+
+export function getSignals(fpd) {
+  const signals = Object.entries({
+    IAB_AUDIENCE_1_1: getSegments(fpd, ['user.data'], 4),
+    IAB_CONTENT_2_2: getSegments(fpd, CLIENT_SECTIONS.map(section => `${section}.content.data`), 6)
+  }).map(([taxonomy, values]) => values.length ? {taxonomy, values} : null)
+    .filter(ob => ob);
+
+  return signals;
+}
+
+export function getSignalsArrayByAuctionsIds(auctionIds) {
```

Comment:
(regarding the test failure) my preferred method of stubbing is dependency injection. E.g. if this did something like `getSignalsArrayByAuctionIds(auctionIds, index = auctionManager.index)`, tests can swap in a mock version of `index`, and I find it a bit easier to reason about compared to sinon's stubs. To do the same with sinon I think you need an extra layer of indirection, basically instead of making `index` a parameter make it a property of some global object pointing to `auctionManager.index`, which you can then stub.